src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -176,6 +103,16 @@ public void AddPeer(IPEndPoint endPoint, IPAddress source)
             this.peers.TryAdd(peerToAdd.Endpoint, peerToAdd);
         }
 
+        public void RemovePeer(IPEndPoint endPoint, IPAddress source)
+        {
+            if (!endPoint.Address.IsRoutable(true))
+                return;
+
+            PeerAddress peer = PeerAddress.Create(endPoint, source);
+            this.peers.TryRemove(peer.Endpoint, out PeerAddress address);
```

Comment:
Why do you create peer address object when you only used endpoint from which you already had before

Reply:
because doing this I go through the same steps as the AddPeer method which ensure that the IP is IPV6

Reply:
That seemed to be quite a bad argument. How about you just make sure yourself?